chromium_src/ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/https_upgrade_exceptions/browser/https_upgrade_exceptions_service.h"
+#include "brave/ios/browser/application_context/brave_application_context_impl.h"
+#import "ios/chrome/browser/shared/model/prefs/pref_names.h"
+#import "ios/components/security_interstitials/https_only_mode/feature.h"
+#import "ios/components/security_interstitials/https_only_mode/https_upgrade_service.h"
+#include "net/base/features.h"
+#include "net/base/url_util.h"
+
+namespace {
+bool CanUpgradeToHTTPS(const GURL& url) {
+  // FIXME: Move this impl out of the chromium_src override
+  BraveApplicationContextImpl* braveContext =
+      static_cast<BraveApplicationContextImpl*>(GetApplicationContext());
+  return braveContext->https_upgrade_exceptions_service()->CanUpgradeToHTTPS(
+      url);
+}
+}  // namespace
+
+// Add checks for Brave-by-default feature flag, standard HTTPS upgrades pref
+// and the brave https upgrade exception list when determining if the navigation
+// should be upgraded
+#define kHttpsUpgrades kHttpsUpgrades) && \
+  !base::FeatureList::IsEnabled(net::features::kBraveHttpsByDefault) && \
```

Comment:
this feature check should be inside `CanUpgradeToHTTPS`

Reply:
The `kHttpsUpgrades` override doesn't call `CanUpgradeToHTTPS`

Reply:
isn't it checked here? https://github.com/brave/brave-core/pull/26957/files#diff-9e16f61dee4e3c4f430cf50c177ab89da6db40727a5ca2e8fb7f13b3c0788a44R19

Reply:
I guess Im a bit confused here:
- We're overriding this spot to add a check for our feature flag since Chromium's is off by default: https://source.chromium.org/chromium/chromium/src/+/main:ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm;l=275
- `IsLocalhost` override below is what calls `CanUpgradeToHTTPS` above, which is only called after the feature flag check added by the `kHttpsUpgrades` override (same line as linked above)

I could add the check again to `CanUpgradeToHTTPS` but its already happened by the first override

Reply:
To be clear I'm talking about `HttpsUpgradeExceptionsService::CanUpgradeToHttps`, but the point is that making the feature check outside of `CanUpgradeToHTTPS` is fragile. If the feature flag is false then `HttpsUpgradeExceptionsService::CanUpgradeToHttps` return return false.